src/ln/channel.rs

```diff
@@ -486,6 +535,31 @@ impl Channel {
 
 		let background_feerate = fee_estimator.get_est_sat_per_1000_weight(ConfirmationTarget::Background);
 
+		let our_dust_limit_satoshis = Channel::derive_our_dust_limit_satoshis(background_feerate);
+		if msg.channel_reserve_satoshis < our_dust_limit_satoshis {
+			send_error_message!("channel_reserve_satoshis too small");
+		}
+		let our_channel_reserve_satoshis = Channel::get_our_channel_reserve_satoshis(msg.funding_satoshis);
+		if our_channel_reserve_satoshis < our_dust_limit_satoshis {
+			send_error_message!("Suitalbe channel reserve not found. aborting");
+		}
+		if our_channel_reserve_satoshis < msg.dust_limit_satoshis {
+			send_error_message!("Dust limit too high for our channel reserve");
+		}
+
+		// check if the funder's amount for the initial commitment tx is sufficient
+		// for full fee payment
+		let funders_amount_msat = msg.funding_satoshis * 1000 - msg.push_msat;
+		if funders_amount_msat < background_feerate * COMMITMENT_TX_BASE_WEIGHT {
+			send_error_message!("Insufficient funding amount for initial commitment");
+		}
+
+		let to_local_msat = msg.push_msat;
+		let to_remote_msat = funders_amount_msat - background_feerate * COMMITMENT_TX_BASE_WEIGHT;
+		if to_local_msat <= msg.channel_reserve_satoshis * 1000 && to_remote_msat <= msg.channel_reserve_satoshis * 1000 {
```

Comment:
Hmm, this looks wrong? You're using the remote_channel_reserve_satoshis for both checks but it only applies to our outputs.